src/test/java/htsjdk/samtools/cram/build/CompressionHeaderFactoryTest.java

```diff
@@ -3,110 +3,79 @@
-            Assert.assertFalse(header.encodingMap.get(key).id == EncodingID.NULL);
-        }
-    }
 
     @Test
     public void testAP_delta() {
         boolean sorted = true;
-        CompressionHeader header = new CompressionHeaderFactory().build(new ArrayList<>(), sorted);
-        Assert.assertEquals(header.APDelta, sorted);
+        CompressionHeader header = new CompressionHeaderFactory(new CRAMEncodingStrategy()).createCompressionHeader(new ArrayList<>(), sorted);
+        Assert.assertEquals(header.isAPDelta(), sorted);
 
         sorted = false;
-        header = new CompressionHeaderFactory().build(new ArrayList<>(), sorted);
-        Assert.assertEquals(header.APDelta, sorted);
+        header = new CompressionHeaderFactory(new CRAMEncodingStrategy()).createCompressionHeader(new ArrayList<>(), sorted);
+        Assert.assertEquals(header.isAPDelta(), sorted);
     }
 
     @Test
     public void testGetDataForTag() {
-        final CompressionHeaderFactory factory = new CompressionHeaderFactory();
-        final List<CramCompressionRecord> records = new ArrayList<>();
-        final CramCompressionRecord record = new CramCompressionRecord();
         final int tagID = ReadTag.name3BytesToInt("ACi".getBytes());
         final byte[] data = new byte[]{1, 2, 3, 4};
-        final ReadTag tag = new ReadTag(tagID, data, ValidationStringency.STRICT);
-        record.tags = new ReadTag[]{tag};
-        records.add(record);
 
-        final byte[] dataForTag = factory.getDataForTag(records, tagID);
+        final CRAMRecord cramRecord = CRAMRecordTestHelper.getCRAMRecordWithTags(
+                "rname", 10, 1, 1, 10, new byte[]{'a', 'c', 'g', 't'},2,
+                Collections.singletonList(new ReadTag(tagID, data, ValidationStringency.STRICT)));
+
+        final CompressionHeaderFactory factory = new CompressionHeaderFactory(new CRAMEncodingStrategy());
+
+        final byte[] dataForTag = factory.getDataForTag(Collections.singletonList(cramRecord), tagID);
         Assert.assertEquals(dataForTag, data);
     }
 
     @Test
-    public void test_getBestExternalCompressor() {
-        try {
-            Assert.assertNotNull(CompressionHeaderFactory.getBestExternalCompressor(null));
-            Assert.fail("NPE expected for null data");
-        } catch (final NullPointerException e) {
-
-        }
-        Assert.assertNotNull(CompressionHeaderFactory.getBestExternalCompressor("".getBytes()));
-        Assert.assertNotNull(CompressionHeaderFactory.getBestExternalCompressor("qwe".getBytes()));
+    public void testGetBestExternalCompressor() {
+        final CRAMEncodingStrategy encodingStrategy = new CRAMEncodingStrategy();
+        final CompressionHeaderFactory compressionHeaderFactory = new CompressionHeaderFactory(encodingStrategy);
+        Assert.assertNotNull(compressionHeaderFactory.getBestExternalCompressor("".getBytes()));
+        Assert.assertNotNull(compressionHeaderFactory.getBestExternalCompressor("qwe".getBytes()));
```

Comment:
Should this test case check the actual compressor chosen, and confirm that it is the one that produces the smallest output?